Runtimes/Unity/BepInEx.Unity.IL2CPP/Il2CppInteropManager.cs

```diff
@@ -282,25 +283,26 @@ private static void GenerateInteropAssemblies()
         }
     }
 
-    private static void DownloadUnityAssemblies()
-    {
+    private static void DownloadUnityAssemblies() {
         var unityVersion = UnityInfo.Version;
-        var source =
-            UnityBaseLibrariesSource.Value.Replace("{VERSION}",
-                                                   $"{unityVersion.Major}.{unityVersion.Minor}.{unityVersion.Build}");
-
-        if (!string.IsNullOrEmpty(source))
-        {
-            Logger.LogMessage("Downloading unity base libraries");
-
-            Directory.CreateDirectory(UnityBaseLibsDirectory);
-            Directory.EnumerateFiles(UnityBaseLibsDirectory, "*.dll").Do(File.Delete);
-
+        var version = $"{unityVersion.Major}.{unityVersion.Minor}.{unityVersion.Build}";
+        var source = UnityBaseLibrariesSource.Value.Replace("{VERSION}", version);
+        var uri = new Uri(source);
+        string file = Path.GetFileName(uri.AbsolutePath);
+
+        var baseFolder = Directory.CreateDirectory(UnityBaseLibsDirectory);
+        baseFolder.EnumerateFiles("*.dll").Do(a=>a.Delete());
```

Comment:
Before if UnityBaseLibrariesSource was empty nothing was done. Now it crashes when trying to create the Uri, which can definitely affect someone (also the old files are now always removed). Better to just keep the old behaviour.


```suggestion
        var source = UnityBaseLibrariesSource.Value.Replace("{VERSION}", version);
        if (string.IsNullOrEmpty(source)) return;

        var uri = new Uri(source);
        string file = Path.GetFileName(uri.AbsolutePath);

        var baseFolder = Directory.CreateDirectory(UnityBaseLibsDirectory);
        baseFolder.EnumerateFiles("*.dll").Do(a=>a.Delete());
```